# Лекция 3. Характеристики систем массового обслуживания. Марковские процессы, уравнения Колмогорова-Чемпена.  Процессы размножения-гибели

- Характеристики систем массового обслуживания
- Марковские процессы, уравнения Колмогорова-Чемпена. Переход от системы дифференциальных уравнений к системе линейных алгебраических уравнений
- Процессы размножения-гибели

## Характеритики систем массового обслуживания

<p style='text-align: justify;'>
Кроме характеристик входного потока заявок (среднее количество заявок, поступивших за единицу времени $\lambda$, средняя длина интервала времени между последовательными поступлениями заявок $\frac{1}{\lambda}$),  режим работы системы зависит и от характристик производительности самой системы: числа каналов, быстродействия каждого канала, времени обслуживания одной заявки $\nu$. Последняя величина является важнейшей характеристикой системы обслуживания и может быть как случайной, так и неслучайной величиной. Мы чаще всего будем рассматривать ее как случайную величину с определенной функцией распределения вероятностей:
</p>

$$G(t)=P(\nu<t)$$
Для практики особый интерес представляет случай, когда величина $\nu$ имеет показательное распределение:
$$G(t)=1-e^{-\mu\cdot t}$$
где $\mu$ - величина, обратная среднему времени обслуживания одной заявки, называемая **интенсивностью обслуживания**.

<p style='text-align: justify;'>Показательный закон не является универсальным законом распределения времени обслуживания. Часто время обслуживания описывается, например, законом Эрланга. Однако, пропускная способность и другие характеристики системы массового обслуживания сравнительно мало зависят от вида закона распределения времени обслуживания, а зависят, главным образом от его среднего времени обслуживания одной заявки $M(\nu)=\frac{1}{\mu}$ . Поэтому в теории массового обслуживания чаще всего пользуются допущением, что время обслуживания распределено по показательному закону. Эта гипотеза позволяет сильно упростить математический аппарат, применяемый для решения задач массового обслуживания, и в ряде случаев, получить простые аналитические формулы для характеристик пропускной способности системы.
</p>

Рассмотрим основные характеристики систем массового обслуживания с однородным потоком заявок:
- **Нагрузка системы** - среднее количество поступающих заявок к среднему количеству обрабатываемых заявок:
$$y=\frac{\lambda}{\mu}\tag{1}$$
- **Загрузка системы** - доля среднего времени, в течение которого система обслуживает заявки $T_o$ к общему времени работы системы $T$:
$$\rho=\lim_{T\rightarrow\infty}\frac{T_o}{T}\tag{2}$$
Если речь идет многоканальном обслуживании, то в качестве $T_o$ выбирается усредненное время обслуживания по всем каналам: $T_o=\frac{\sum_{i=1}^K T_i}{K}$ 

- **Коэффициент простоя системы** (величина, противоположная загрузке системы):
$$\eta = 1-\rho\tag{3}$$
- **Вероятность потери заявок** отношение количества потерянных заявок $N_{\text{п}}(T)$ за время $T$ к общему числу заявок, поступившив в систему за время $T$:
$$P_{\text{отк}}=\lim_{T\rightarrow \infty}\frac{N_{\text{п}}(T)}{N(T)}\tag{4}$$

- **Вероятность обслуживания заявок** - величина, противоположная вероятности потери:
$$P_{\text{обсл}}=1-P_{\text{отк}}=\lim_{T\rightarrow\infty}\frac{N_{\text{о}}(T)}{N(T)}\tag{5}$$
где $N_{\text{о}}(T)$ - число обслуженных в системе заявок за время $T$, $N_{\text{о}}(T)+N_{\text{п}}
(T)=N(T)$

- **Абсолютная пропускная способность системы** (производительность системы) - это интенсивность обслуженных заявок, выходящих из системы:
$$\lambda'=P_{\text{обсл}}\lambda=(1-P_{\text{отк}})\lambda\tag{6}$$
В системах с неограниченной очередью $P_{\text{отк}}=0, \lambda'=\lambda$

- **Относительная пропускная способность** (или вероятность обслуживания поступившей заявки):
$$Q=\frac{\lambda'}{\lambda} = P_{\text{обсл}}\tag{6'}$$

- **Интенсивность потока потерянных заявок**: 
$$\lambda''=P_{\text{отк}}\lambda=(1-P_{\text{обсл}})\lambda\tag{7}$$
при $\lambda'+\lambda''=\lambda$

- **Среднее время ожидания заявок в очереди** $W_{\text{оч}}$
- **Среднее время пребывания заявок в системе** $T_{\text{сист}}$ - есть сумма среднего времени ожидания в очереди $W_{\text{оч}}$ и среднего времени обслуживания $\frac{1}{\mu}$:
$$T_{\text{сист}}=W_{\text{оч}}+\frac{1}{\mu}\tag{8}$$

- **Среднее число заявок в системе** $L_{\text{сист}}$:
$$L_{\text{сист}}=\lambda\cdot T_{\text{сист}}\tag{9}$$

- **Средняя длина очереди** $L_{\text{оч}}$:
$$L_{\text{оч}}=\lambda\cdot W_{\text{оч}}\tag{10}$$

<p style='text-align: justify;'>Формулы (9), (10) известны как <b>формулы Литтла</b>: для любой <i>открытой</i> СМО, при любом характере потока заявок, любом распределении времени обслуживания, любой дисциплине обслуживания среднее число заявок в системе равно произведению интенсивности входящего потока на среднее время пребывания заявки в системе; средняя длина очереди равна произведению интенсивности входящего потока  на среднее время ожидания заявки в очереди.</p>

Проведем обоснование справедливости формул Литтла. Обозначим через $S(t)$ - число заявок, пришедших в систему за интервал времени от 0 до $t$, через $E(t)$ - число заявок, получивших обслуживание за интервал времени от 0 до $t$.

![image](./ris11.jpeg)

Из рисунка видно, что в момент времени $t_0$ в систему поступила одна заявка, в момент времени $t_1$ в систему поступила вторая заявка, в $t_2$ - третья, и т.д.:
$$S(t<t_1)=1, S(t_1<t<t_2)=2, S(t_2<t<t_3)=3, S(t_3<t<t_4)=4, \ldots$$

Будем считать, что в интервале времени $[0,t_1']$  $E(t<t_1')=1$, тогда
$$E(t_1'<t<t_2')=2, E(t_2'<t<t_3')=3, \ldots$$
Также имеем $T_1=t_1'-t_1$ - время, в течение которого вторая заявка находилась в системе (ожидала обслуживания и обслуживалась), $T_2 = t_2'-t_2$ - время, в течение которого третья заявка находилась в системе (ожидала обслуживания и обслуживалась), $T_3 = t_3'-t_3$ - время, в течение которого четвертая заявка находилась в системе (ожидала обслуживания и обслуживалась) и т.д.

Пусть за некий интервал времени $\tau$ на вход системы поступит $S(\tau)$ заявок. Тогда средняя интенсивность входящего потока выразится:
$$\lambda = \frac{S(\tau)}{\tau}$$

Рассмотрим функцию $U(t)=S(t)-E(t)$ - число заявок, ожидающих обслуживания или находящихся на обслуживании в момент времени $t$, тогда интеграл 
$\int_0^\tau U(t) dt=\sum T_i$ определяет суммарное время нахождения заявок в системе. Данный интеграл можно представить суммой прямоугольников (на рисунке затенённые области), у которых вертикальная сторона равна 1, горизонтальная равна $T_i$.

Среднее время нахождения заявок в системе за интервал времени $\tau$:
$$T_{\text{сист}}=\frac{\sum T_i}{S(\tau)}=\frac{\sum T_i}{\lambda\cdot \tau}$$

Среднее число заявок в системе:
$$L_{\text{сист}}=\frac{\int_0^\tau U(t) dt}{\tau}=\frac{\sum T_i}{\tau}\Rightarrow L_{\text{сист}}=\lambda\cdot T_{\text{сист}}$$

Аналогичным образом обосновывается и формула (10), так из графика можно заметить, что время ожидания 4 заявки $W_4 = t_2'-t_3$, далее рассуждения аналогичны.

Ещё раз подчеркнем, формулы (9), (10) носят универсальный характер для любой **открытой** СМО (одноканальной, многоканальной, при любых видах потоков заявок и обслуживании), единственное требование к потокам заявок и обслуживания - стационарность.

- Для **открытых** СМО универсальное значение имеет и формула, выражающая **среднее число занятых каналов** через пропускную способность $\lambda$:
$y=\frac{\lambda}{\mu}\tag{10'}$

В завершении отметим, что в общем случае все рассмотренные характеристики зависят от времени. Но многие СМО работают с неизменяемыми характеристиками продолжительное время, и для них успевают установиться стационарные режимы, для которых характерны так называемые финальные характеристики эффективности СМО.

## Марковские процессы. Уравнения Колмогорова-Чемпена. Переход от системы дифференциальных уравнений к системе линейных алгебраических уравнений.

### Марковские процессы

Понятие случайного процесса является обобщением понятия случайной величины.

**Определение 1.** *Случайным процессом $X(t)$ называется функция, значение которой при любом значении аргумента $t$ является случайной величиной.*

При фиксированной $t=t_0$ (сечение случайного процесса в момент $t=t_0$) $X(t_0)$ представляет собой обычную случайную величину.

Примеры случайных процессов:
- Численность населения города в течение времени;
- Число заявок, поступающих в ремонтную службу фирмы, с течением времени.



Случайный процесс можно  записать в виде функции двух переменных $X(t,w)$, где $w\in \Omega$, $w$ - элементарное событие, $\Omega$ - пространство элементарных событий, $t\in T$, $T$ - множество значений аргумента $t$.

<p style='text-align: justify;'>Среди всех случайных процессов особую роль в теории массового обслуживания играют так называемые марковские случайные процессы. Марковские случайные процессы названы по имени выдающегося русского математика А. А. Маркова (1856–1922), впервые начавшего изучение вероятностной связи случайных величин.</p>

**Определение 2.** *Случайный процесс $X(t)$ называется **марковским**, если он обладает свойством *отсутствия последействия*: для любого момента времени $t_0$ вероятностные характеристики процесса в будущем (при $t>t_0$) зависят только от его состояния в данный момент времени $t_0$ и не зависят от того, по какой траектории процесс пришел в это состояние.* 

**Определение 3.** Марковский случайный процесс $X(t)$ называется **цепью Маркова**, если множество его состояний дискретно (конечно или счетно), а изменение состояния (переход из одного состояния в другое) происходит мгновенно (скачком) в определенные моменты времени $t_0, t_1, t_2, \ldots$, называемые шагами процесса (в случае процесса с дискретным временем) или в случайные моменты времени (для процесса с непрерывным временем).

Поскольку марковская цепь является частным случаем марковских случайных процессов, для каждого шага вероятность перехода из состояния $S_i$ в состояние $S_j$, обозначенная как $p_{ij}$ не зависит от того, когда и как система пришла в состояние $S_i$.

**Определение 4.** Цепь Маркова называется **однородной**, если вероятности переходов $p_{ij}$ не зависят от номера шага $\forall: t_k$, в противном случае, если $p_{ij}=p_{ij}(t_k)$, то цепь Маркова называется **неоднородной**.

Использование вероятности перехода $p_{ij}$ возможно только для так называемых **процессов с дискретным временем** - изменение состояний происходит только в определенные моменты времени $t_1, t_2, \ldots$.

Для случайных процессов с **непрерывным временем** время между переходами из одного состояния в другое случайно. Это означает, что вероятность перехода из одного состояния в другое не может быть задана, поскольку вероятность такого перехода точно в произвольный момент времени $t$ равна нулю. Для описания переходов между состояниями случайного процесса с непрерывным временем вместо вероятностей переходов вводится параметр, называемый интенсивностью перехода $\lambda$.

Вероятности переходов удобно задавать в виде квадратной матрицы:
$$P=[p_{ij}|i,j=1,\ldots,n]$$
при выполнении условий *нормировки*:
$$0\leq p_{ij}\leq 1; \sum_{j=1}^{n}p_{ij}=1\tag{11}$$

Матрица $P$ называется **стохастической**, ее можно использовать для вычисления вероятностей нахождения в каждом из $n$ состояний через $k$ моментов времени.

Обозначим через $p(0)=(p_1(0),p_2(0), \ldots, p_n(0))$ - вектор вероятностей нахождения в каждом из $n$ состояний в начальный момент времени $t=0$.

Тогда $$p(1)=p(0)\cdot P$$
$$p(2)=p(1)\cdot P=p(0)\cdot P^2$$
$$\cdots$$
$$p(k)=p(0)\cdot P^k\tag{12}$$
Рассмотрим пример.

**Пример 1.**

Система, может находиться в трех различных состояниях с переходными вероятностями $p_{ij}$, указанными на рисунке:
<p align="center" width="100%">
  <img width="33%" src="ris1.jpeg"/>
</p>

Начальный вектор вероятностей $p(0)=(0.2, 0.3, 0.5)$. Найти вероятности находится в каждом из заданных состояний через $k$ шагов, где $k$ - заданное число, или $k\rightarrow\infty$.

**Решение**

Исходя из рисунка матрица переходных вероятностей имеет следующий вид:
\begin{equation}
P = \begin{pmatrix} 0 & 0.3 & 0.7 \\ 0.1 & 0.5 & 0.4 \\0.2 & 0.8 & 0 \end{pmatrix} 
\end{equation}
Тогда 
$$p(k)=(0.2, 0.3, 0.5)\cdot \begin{pmatrix} 0 & 0.3 & 0.7 \\ 0.1 & 0.5 & 0.4 \\0.2 & 0.8 & 0 \end{pmatrix}^k$$

Проведем вычисления для некоторых конкретных значений $k$:

In [1]:
k<-3
P<-matrix(c(0,0.3,0.7,0.1,0.5,0.4,0.2,0.8,0), ncol=3, nrow=3, byrow=TRUE)
p0<-matrix(c(0.2, 0.3, 0.5), ncol=3, nrow=1, byrow=TRUE)

In [2]:
print(p0)
print(P)

     [,1] [,2] [,3]
[1,]  0.2  0.3  0.5
     [,1] [,2] [,3]
[1,]  0.0  0.3  0.7
[2,]  0.1  0.5  0.4
[3,]  0.2  0.8  0.0


In [12]:
#install.packages("matrixcalc")

Устанавливаю пакет в ‘/home/juna/R/x86_64-pc-linux-gnu-library/4.2’
(потому что ‘lib’ не определено)



In [3]:
library(matrixcalc)
p0 %*% matrix.power(P,k)

[,1]   [,2]   [,3]  
[1,] 0.1222 0.5779 0.2999

In [4]:
k<-9
p0 %*% matrix.power(P,k)

[,1]      [,2]      [,3]     
[1,] 0.1192415 0.5696376 0.3111209

In [5]:
k<-20
p0 %*% matrix.power(P,k)

[,1]      [,2]      [,3]     
[1,] 0.1192053 0.5695364 0.3112583

Попробуем задать другой начальный вектор вероятностей:

In [6]:
k<-20
matrix(c(0.1, 0.8, 0.1), ncol=3, nrow=1, byrow=TRUE) %*% matrix.power(P,k)

[,1]      [,2]      [,3]     
[1,] 0.1192053 0.5695365 0.3112582

Видно, что с увеличением количества шагов, вероятности нахождения в заданном состоянии практически не меняются и они не зависят от начальных условий. Такие вероятности называются **установившимися**, они характеризуют марковскую цепь при $t\rightarrow\infty$.

Их можно найти, исходя из следующих условий:
$$Q=q(0)\cdot P^k, k\rightarrow\infty\Rightarrow Q=Q\cdot P$$
$$Q(P-E)=0$$
где $E$ - единичная матрица, 0 - нулевая вектор-строка.
$$\begin{pmatrix} q_1 & q_2 & q_3 \end{pmatrix}\cdot\left(\begin{pmatrix} 0 & 0.3 & 0.7 \\ 0.1 & 0.5 & 0.4 \\0.2 & 0.8 & 0 \end{pmatrix}-\begin{pmatrix} 1 & 0 & 0 \\ 0 & 1 & 0 \\0 & 0 & 1 \end{pmatrix}\right)=\begin{pmatrix} 0 & 0 & 0 \end{pmatrix}$$
$$\begin{pmatrix} q_1 & q_2 & q_3 \end{pmatrix}\cdot\begin{pmatrix} -1 & 0.3 & 0.7 \\ 0.1 & -0.5 & 0.4 \\0.2 & 0.8 & -1 \end{pmatrix}=\begin{pmatrix} 0 & 0 & 0 \end{pmatrix}$$

$$\begin{cases} -q_1 + 0.1q_2 + 0.2q_3=0 \\ 0.3q_1 - 0.5q_2 + 0.8q_3=0 \\ 0.7q_1+0.4q_2-q_3=0 \end{cases}$$


In [7]:
det(matrix(c(-1,0.1,0.2,0.3,-0.5,0.8,0.7,0.4,-1),ncol=3, nrow=3, byrow=TRUE))

[1] 1.04361e-16

Полученная система уравнений будет зависима, поэтому для нахождения вектора $(q_1,q_2,q_3)$ вместо одного из этих уравнений можно подставить условие нормировки: $q_1+q_2+q_3=1$
$$\begin{pmatrix} q_1 & q_2 & q_3 \end{pmatrix}\cdot\begin{pmatrix} -1 & 0.3 & 1 \\ 0.1 & -0.5 & 1 \\0.2 & 0.8 & 1 \end{pmatrix}=\begin{pmatrix} 0 & 0 & 1 \end{pmatrix}$$
$$\begin{pmatrix} q_1 & q_2 & q_3 \end{pmatrix}=\begin{pmatrix} 0 & 0 & 1 \end{pmatrix}\cdot \begin{pmatrix} -1 & 0.3 & 1 \\ 0.1 & -0.5 & 1 \\0.2 & 0.8 & 1 \end{pmatrix}^{-1}$$

In [8]:
res<-matrix(c(0, 0, 1), ncol=3, nrow=1, byrow=TRUE) %*% 
solve(matrix(c(-1,0.3,1,0.1,-0.5,1,0.2,0.8,1), ncol=3, nrow=3, byrow=TRUE))
res

[,1]      [,2]      [,3]     
[1,] 0.1192053 0.5695364 0.3112583

Как видно, результаты совпадают с численным экспериментом. В целом стационарные вероятности $q_i$ равны доле шагов, на которых система находилась в состоянии $i$. Если найди величины, обратные вероятностям, то они имеют смысл среднего времени возвращения в состояние $i$ (среднее число шагов, через которое мы опять вернемся в состояние $i$).

In [5]:
1/res

[,1]     [,2]     [,3]    
[1,] 8.388889 1.755814 3.212766

Рассмотрим еще ряд определений и свойств.

**Определение 5.** Состояние $i$ называется **несущественным**, если существует такое состояние $j$, в которое система может перейти за конечное число шагов $n$, но не может вернуться в $i$-е ни за какое число шагов:
$$p_{ij}(n)>0, \forall m: p_{ji}(m)=0$$

Все остальные состояния **существенные**.

**Определение 6.** Состояния $i$ и $j$ называются **сообщающимися**, если существуют $n$ и $k$ такие, что
$$p_{ij}(n)>0, p_{ji}(k)>0$$

**Определение 7.** Все существенные состояния можно разбить на классы, которые состоят из сообщающихся состояний, и ни из одного состояния данного класса нельзя перейти в состояние другого класса. Такие классы называют **неразложимыми**, или **замкнутыми**.


### Уравнения Колмогорова

Рассмотрим теперь марковские процессы с непрерывным временем. Пусть некоторое техническое устройство состоит из двух узлов, каждый из которых в случайный момент времени может выйти из строя, после чего мгновенно начинается ремонт узла, тоже продолжающийся случайное время. Можно выделить следующие состояния системы:
- $S_0$ - оба узла исправны;
- $S_1$ - первый узел ремонтируется, второй исправен;
- $S_2$ - второй узел ремонтируется, первый исправен;
- $S_3$ - оба узла ремонтируются.

Переходы системы из состояния в состояние происходят практически мгновенно, в случайные моменты выхода из строя того или другого узла или окончания ремонта. На рисунке ниже представлен граф состояний системы.
Стрелка, направленная из $S_0$ в $S_1$, означает переход в момент отказа первого узла; стрелка, направленная обратно, из $S_1$ в $S_0$ - переход в момент окончания ремонта этого узла. Остальные стрелки объясняются аналогично.

<img src="ris2.jpeg">

Обозначим на схеме $a_i=\lambda_i$ - интенсивность поломки $i$-го узла; $m_i=\mu_i$ - интенсивность ремонта $i$-го узла. Будем исходить из того, что все потоки поломки, ремонта являются простейшими потоками.

Вероятность $S_i$-го состояния в момент времени $t$ обозначим через $P_i(t)$. Ясно, что для любого момента времени сумма всех вероятностей равна единице:
 $$\forall t:\sum_{i=0}^nP_i(t)=1\tag{13}$$

Для нахождения вероятностей как функций времени составляются и решаются *уравнения Колмогорова - Чемпена* - дифференциальные уравнения, в которых неизвестными функциями являются вероятности состояний. 

Составим эти уравнения для заданной системы. Рассмотрим, например, вероятность $P_0(t)$. Это вероятность того, что в момент времени $t$ система будет находиться в состоянии $S_0$. Придадим $t$ малое приращение $\Delta t$, тогда $P_0(t+\Delta t)$ - это вероятность, что за время $\Delta t$ система будет в состоянии $S_0$. Это может случиться в трех случаях:
- система уже была в состоянии $S_0$ и за время $\Delta t$ не вышла из него;
- система находилась в состоянии $S_1$ и за время $\Delta t$ перешла в состояние $S_0$;
- система находилась в состоянии $S_2$ и за время $\Delta t$ перешла в состояние $S_0$.

В первом случае вероятность того, что в момент времени $t$ система была в состоянии $S_0$ равна $P_0(t)$.
Эту вероятность нужно умножить на вероятность того, что за время $\Delta t$ система не перейдет в другое состояние. Суммарный поток событий, выводящий систему из состояния $S_0$, будет простейшим с интенсивностью $\lambda_1+\lambda_2$.

Значит, вероятность того, что за время $\Delta t$ система выйдет из состояния $S_0$ равна $(\lambda_1+\lambda_2)\cdot \Delta t$, вероятность того, что не выйдет равна $(1-(\lambda_1+\lambda_2)\Delta t)$. Тогда вероятность первого варианта равна $P_0(t)\cdot (1-(\lambda_1+\lambda_2)\Delta t)$. Вероятность второго варианта равна $\mu_1\cdot \Delta t\cdot P_1(t)$, вероятность третьего варианта $\mu_2\cdot \Delta t\cdot  P_2(t)$. Складывая вероятности всех вариантов, получим:
$$P_0(t+\Delta t)=P_0(t)\cdot (1-(\lambda_1+\lambda_2)\Delta t)+\mu_1\Delta t P_1(t)+\mu_2 \Delta t P_2(t)$$

Проведем перегруппировку и разделим обе части на $\Delta t$:
$$\frac{P_0(t+\Delta t)-P_0(t)}{\Delta t}=-P_0(t)\cdot (\lambda_1+\lambda_2)+P_1(t)\mu_1+P_2(t)\mu_2$$

Устремим $\Delta t$ к нулю, получим:
$$\frac{dP_0(t)}{dt}=-P_0(t)\cdot (\lambda_1+\lambda_2)+P_1(t)\mu_1+P_2(t)\mu_2\tag{14}$$

Аналогично составляются остальные уравнения:
$$\frac{dP_1(t)}{dt}=P_0(t)\cdot \lambda_1-P_1(t)(\mu_1+\lambda_2)+P_3(t)\mu_2\tag{15}$$
$$\frac{dP_2(t)}{dt}=P_0(t)\cdot \lambda_2-P_2(t)(\mu_2+\lambda_1)+P_3(t)\mu_1\tag{16}$$
$$\frac{dP_3(t)}{dt}=P_1(t)\cdot \lambda_2+P_2(t)\lambda_1-P_3(t)(\mu_1+\mu_2)\tag{17}$$

Полученные дифференциальные уравнения (14), (15), (16), (17) и называются **уравнениями Колмогорова-Чемпена**.

Для удобства их составления можно использовать следующее **правило**: *в левой части каждого уравнения стоит производная вероятности данного состояния, в правой части - сумма произведений вероятностей всех состояний, из которых идут стрелки в данное состояние, на интенсивности соответствующих потоков, входящих в состояние, минус суммарная интенсивность всех потоков, выводящих систему из данного состояния, умноженная на вероятность данного состояния.*

Решая совместную систему дифференциальных уравнений (14), (15), (16), (17), можно выразить динамические вероятности нахождения системы в заданном состоянии как функцию времени.

### Переход от системы дифференциальных уравнений к системе линейных алгебраических уравнений.

По аналогии с марковскими процессами дискретного времени, можно поставить вопрос: что будет происходить с вероятностями при $t\rightarrow\infty$, будут ли эти вероятности $P_i(t)$ стремиться к каким-либо пределам.

Если эти вероятности существуют и не зависят от начальных условий, то они называются **финальными вероятностями состояний**.

Для решения системы дифференциальных уравнений: 
$$\begin{cases} \frac{dP_0(t)}{dt}=-P_0(t)\cdot (\lambda_1+\lambda_2)+P_1(t)\mu_1+P_2(t)\mu_2\\
\frac{dP_1(t)}{dt}=P_0(t)\cdot \lambda_1-P_1(t)(\mu_1+\lambda_2)+P_3(t)\mu_2\\
\frac{dP_2(t)}{dt}=P_0(t)\cdot \lambda_2-P_2(t)(\mu_2+\lambda_1)+P_3(t)\mu_1\\
\frac{dP_3(t)}{dt}=P_1(t)\cdot \lambda_2+P_2(t)\lambda_1-P_3(t)(\mu_1+\mu_2)\\
\end{cases}\tag{18}$$ 
удобно использовать преобразование Лапласа:
$$\begin{cases} s\cdot P_0(s)=-P_0(s)\cdot (\lambda_1+\lambda_2)+P_1(s)\mu_1+P_2(s)\mu_2\\
s\cdot P_1(s)=P_0(s)\cdot \lambda_1-P_1(s)(\mu_1+\lambda_2)+P_3(s)\mu_2\\
s\cdot P_2(s)=P_0(s)\cdot \lambda_2-P_2(s)(\mu_2+\lambda_1)+P_3(s)\mu_1\\
s\cdot P_3(s)=P_1(s)\cdot \lambda_2+P_2(s)\lambda_1-P_3(s)(\mu_1+\mu_2)\\
\end{cases}$$ 

$$\begin{cases} -P_0(s)\cdot (\lambda_1+\lambda_2+s)+P_1(s)\mu_1+P_2(s)\mu_2=0\\
P_0(s)\cdot \lambda_1-P_1(s)(\mu_1+\lambda_2+s)+P_3(s)\mu_2=0\\
P_0(s)\cdot \lambda_2-P_2(s)(\mu_2+\lambda_1+s)+P_3(s)\mu_1=0\\
P_1(s)\cdot \lambda_2+P_2(s)\lambda_1-P_3(s)(\mu_1+\mu_2+s)=0\\
\end{cases}$$ 

Для решения данной системы вместо любого уравнения нужно добавить уравнение нормировки (13):
$$P_0(s)+P_1(s)+P_2(s)+P_3(s)=\frac{1}{s}$$

$$\begin{pmatrix} P_0(s) & P_1(s) & P_2(s) & P_3(s) \end{pmatrix}\cdot\begin{pmatrix} -(\lambda_1+\lambda_2+s) & \mu_1 & \mu_2 & 0 \\ \lambda_1 & -(\mu_1+\lambda_2+s) & 0 & \mu_2 \\ \lambda_2 & 0& -(\mu_2+\lambda_1+s) & \mu_1 \\ 1 & 1 & 1 & 1 \end{pmatrix}^T=\begin{pmatrix} 0 & 0 & 0 & \frac{1}{s} \end{pmatrix}$$
$$\begin{pmatrix} P_0(s) & P_1(s) & P_2(s) & P_3(s) \end{pmatrix}=\begin{pmatrix} 0 & 0 & 0 & \frac{1}{s} \end{pmatrix}\cdot\left(\begin{pmatrix} -(\lambda_1+\lambda_2+s) & \mu_1 & \mu_2 & 0 \\ \lambda_1 & -(\mu_1+\lambda_2+s) & 0 & \mu_2 \\ \lambda_2 & 0& -(\mu_2+\lambda_1+s) & \mu_1 \\ 1 & 1 & 1 & 1 \end{pmatrix}^T\right)^{-1}$$

Зададимся в качестве примера численными значениями $\lambda_1=0.3, \lambda_2=0.1, \mu_1=0.8, \mu_2=0.6$.

Для выполнения символьных вычислений используем maxima.

In [1]:
matrix([0,0,0,1/s]);

                                [          1 ]
(%o0)                           [ 0  0  0  - ]
                                [          s ]

In [8]:
a1:0.3$ a2:0.1$m1:0.8$m2:0.6$
f:matrix([-(a1+a2+s),m1,m2,0],[a1,-(m1+a2+s),0,m2],[a2,0,-(m2+a1+s),m1],[1,1,1,1]);

         [ (- s) - 0.4      0.8                 0.6               0  ]
         [                                                           ]
         [     0.3      (- s) - 0.9              0               0.6 ]
(%o21)   [                                                           ]
         [     0.1           0       (- s) - 0.8999999999999999  0.8 ]
         [                                                           ]
         [      1            1                   1                1  ]

In [7]:
f[1];

(%o16)                    [(- s) - 0.4, 0.3, 0.1, 1]

In [9]:
p0:expand(matrix([0,0,0,1/s]).invert(f)[1]);
p1:expand(matrix([0,0,0,1/s]).invert(f)[2]);
p2:expand(matrix([0,0,0,1/s]).invert(f)[3]);
p3:expand(matrix([0,0,0,1/s]).invert(f)[4]);

                        0.96 s
(%o22) (- -----------------------------------)
              4         3         2
          (- s ) - 3.6 s  - 4.01 s  - 1.386 s
                                                    0.8639999999999999
                                          - -----------------------------------
                                                4         3         2
                                            (- s ) - 3.6 s  - 4.01 s  - 1.386 s

                             2
                        0.6 s
(%o23) (- -----------------------------------)
              4         3         2
          (- s ) - 3.6 s  - 4.01 s  - 1.386 s
             0.7799999999999999 s                          0.324
    - ----------------------------------- - -----------------------------------
          4         3         2                 4         3         2
      (- s ) - 3.6 s  - 4.01 s  - 1.386 s   (- s ) - 3.6 s  - 4.01 s  - 1.386 s

                             2
                        0.8 s
(%o24) (- -----------------------------------)
              4         3         2
          (- s ) - 3.6 s  - 4.01 s  - 1.386 s
                    1.04 s                                 0.144
    - ----------------------------------- - -----------------------------------
          4         3         2                 4         3         2
      (- s ) - 3.6 s  - 4.01 s  - 1.386 s   (- s ) - 3.6 s  - 4.01 s  - 1.386 s

                           3
                          s
(%o25) (- -----------------------------------)
              4         3         2
          (- s ) - 3.6 s  - 4.01 s  - 1.386 s
                      2
                 2.2 s                                 1.23 s
 - ----------------------------------- - -----------------------------------
       4         3         2                 4         3         2
   (- s ) - 3.6 s  - 4.01 s  - 1.386 s   (- s ) - 3.6 s  - 4.01 s  - 1.386 s
           0.05400000000000005
 - -----------------------------------
       4         3         2
   (- s ) - 3.6 s  - 4.01 s  - 1.386 s

Таким образом, для расчетного примера получили:
$$P_0(s)=\frac{0.96s+0.864}{s(s^3+3.6s^2+4.01s+1.386)}$$
$$P_1(s)=\frac{0.6s^2+0.78s+0.324}{s(s^3+3.6s^2+4.01s+1.386)}$$
$$P_2(s)=\frac{0.8s^2+1.04s+0.144}{s(s^3+3.6s^2+4.01s+1.386)}$$
$$P_3(s)=\frac{s^3+2.2s^2+1.23s+0.054}{s(s^3+3.6s^2+4.01s+1.386)}$$

Чтобы получить вероятности как функции времени, выполним обратное преобразование Лапласа:

In [10]:
ilt(p0,s,t);
ilt(p1,s,t);
ilt(p2,s,t);
ilt(p3,s,t);


rat: replaced -0.8639999999999999 by -108/125 = -0.864

rat: replaced -1.386 by -693/500 = -1.386

rat: replaced -4.01 by -401/100 = -4.01

rat: replaced -3.6 by -18/5 = -3.6

rat: replaced -0.96 by -24/25 = -0.96

rat: replaced -1.386 by -693/500 = -1.386

rat: replaced -4.01 by -401/100 = -4.01

rat: replaced -3.6 by -18/5 = -3.6


                          7 t           11 t          9 t
                        - ---         - ----        - ---
                          10             10            5
                   48 %e         48 %e         48 %e        48
(%o26)          (- ----------) - ----------- + ---------- + --
                       77            77            77       77


rat: replaced -0.324 by -81/250 = -0.324

rat: replaced -1.386 by -693/500 = -1.386

rat: replaced -4.01 by -401/100 = -4.01

rat: replaced -3.6 by -18/5 = -3.6

rat: replaced -0.7799999999999999 by -39/50 = -0.78

rat: replaced -1.386 by -693/500 = -1.386

rat: replaced -4.01 by -401/100 = -4.01

rat: replaced -3.6 by -18/5 = -3.6

rat: replaced -0.6 by -3/5 = -0.6

rat: replaced -1.386 by -693/500 = -1.386

rat: replaced -4.01 by -401/100 = -4.01

rat: replaced -3.6 by -18/5 = -3.6


                          7 t           11 t          9 t
                        - ---         - ----        - ---
                          10             10            5
                   18 %e         48 %e         48 %e        18
(%o27)          (- ----------) + ----------- - ---------- + --
                       77            77            77       77


rat: replaced -0.144 by -18/125 = -0.144

rat: replaced -1.386 by -693/500 = -1.386

rat: replaced -4.01 by -401/100 = -4.01

rat: replaced -3.6 by -18/5 = -3.6

rat: replaced -1.04 by -26/25 = -1.04

rat: replaced -1.386 by -693/500 = -1.386

rat: replaced -4.01 by -401/100 = -4.01

rat: replaced -3.6 by -18/5 = -3.6

rat: replaced -0.8 by -4/5 = -0.8

rat: replaced -1.386 by -693/500 = -1.386

rat: replaced -4.01 by -401/100 = -4.01

rat: replaced -3.6 by -18/5 = -3.6


                          7 t         11 t          9 t
                        - ---       - ----        - ---
                          10           10            5
                   48 %e        8 %e         48 %e        8
(%o28)             ---------- - ---------- - ---------- + --
                       77           77           77       77


rat: replaced -0.05400000000000005 by -27/500 = -0.054

rat: replaced -1.386 by -693/500 = -1.386

rat: replaced -4.01 by -401/100 = -4.01

rat: replaced -3.6 by -18/5 = -3.6

rat: replaced -1.23 by -123/100 = -1.23

rat: replaced -1.386 by -693/500 = -1.386

rat: replaced -4.01 by -401/100 = -4.01

rat: replaced -3.6 by -18/5 = -3.6

rat: replaced -2.2 by -11/5 = -2.2

rat: replaced -1.386 by -693/500 = -1.386

rat: replaced -4.01 by -401/100 = -4.01

rat: replaced -3.6 by -18/5 = -3.6

rat: replaced -1.386 by -693/500 = -1.386

rat: replaced -4.01 by -401/100 = -4.01

rat: replaced -3.6 by -18/5 = -3.6


                          7 t         11 t          9 t
                        - ---       - ----        - ---
                          10           10            5
                   18 %e        8 %e         48 %e        3
(%o29)             ---------- + ---------- + ---------- + --
                       77           77           77       77

Из полученных решений в частности следует, что финальные вероятности состояний рассматриваемого примера равны:
$$P_0(t)_{t\rightarrow\infty}=\frac{48}{77}$$
$$P_1(t)_{t\rightarrow\infty}=\frac{18}{77}$$
$$P_2(t)_{t\rightarrow\infty}=\frac{8}{77}$$
$$P_3(t)_{t\rightarrow\infty}=\frac{3}{77}$$

In [11]:
48/77+18/77+8/77+3/77;

(%o30)                                 1

Конечно, финальные вероятности можно искать и без решения дифференциальных уравнений, положив левую часть всех уравнений системы (18) равной нулю (если вероятности не меняются, их производные равны нулю).
$$\begin{cases} -P_0(t)\cdot (\lambda_1+\lambda_2)+P_1(t)\mu_1+P_2(t)\mu_2=0\\
P_0(t)\cdot \lambda_1-P_1(t)(\mu_1+\lambda_2)+P_3(t)\mu_2=0\\
P_0(t)\cdot \lambda_2-P_2(t)(\mu_2+\lambda_1)+P_3(t)\mu_1=0\\
P_1(t)\cdot \lambda_2+P_2(t)\lambda_1-P_3(t)(\mu_1+\mu_2)=0\\
\end{cases}\tag{19}$$ 
Добавив сюда уравнение нормировки (13), можно найти решение.

## Процессы размножения, гибели

Рассмотрим марковский процесс с непрерывным временем под названием модель "гибели и размножения", которая
часто встречается в разнообразных практических задачах. 

Своим названием эта модель обязана биологической задаче об изменении численности популяции и распространении эпидемий, которая формулируется следующим образом: рассмотрим развитие некоторой популяции, особи которой могут рождаться и умирать. Положим, что при наличии $i$ особей в популяции рождение новых особей происходит с интенсивностью $\lambda_i$ , а с интенсивностью $\mu_i$ особи умирают.

Особенностью графа состояний для схемы рождения, гибели является то, что все состояния системы можно вытянуть в одну цепочку, в которой каждое из средних состояний $S_1, S_2, \ldots, S_{n-1}$ связано прямой и обратной стрелкой с каждым из соседних состояний - правым и левым, а крайние состояния $S_0, S_n$ - только с одним соседним состоянием.

<img src="ris3.jpeg">

Пусть в любой момент времени может происходить рождение или гибель только одной особи, и интервалы времени между двумя моментами рождения и гибели распределены по экспоненциальному закону с параметрами $\lambda_i$ и $\mu_i$ соответственно. Тогда процесс «гибели и размножения» может быть представлен марковским случайным процессом с непрерывным временем, в котором состояние $S_i$ соответствует наличию $i$ особей в популяции,
причем число состояний может быть конечным или бесконечным. Про этом состояние $S_0$ соответствует вырождению популяции.

Пользуясь графом состояний, запишем систему линейных алгебраических уравнений для финальных вероятностей.

Для нулевого узла:
$$P_0\lambda_1=P_1\mu_1\tag{20}$$
Для первого узла:
$$P_1(\mu1+\lambda_2)=P_0\lambda_1+P_2\mu_2\tag{21}$$
Для $k$-го узла:
$$P_k(\mu_k+\lambda_{k+1})=P_{k-1}\lambda_{k}+P_{k+1}\mu_{k+1}\tag{22}$$
Для $n$-го узла:
$$P_n\mu_n=P_{n-1}\lambda_n\tag{23}$$

$$P_1=\frac{\lambda_1}{\mu_1}\cdot P_0$$
$$P_2=P_1\frac{(\mu_1+\lambda_2)}{\mu_2}-P_0\frac{\lambda_1}{\mu_2}=P_0\left(\frac{\mu_1+\lambda_2}{\mu_2}\cdot \frac{\lambda_1}{\mu_1}-\frac{\lambda_1}{\mu_2}\right)$$

$$P_2=P_0\cdot \frac{\lambda_1\lambda_2}{\mu_1\mu_2}$$
Аналогично выражается:
$$P_3=P_0\cdot \frac{\lambda_1\lambda_2\lambda_3}{\mu_1\mu_2\mu_3}$$

И вообще, для любого $k=1,2,3,\ldots, n$:
$$P_k=\frac{\prod_{i=1}^k \lambda_i}{\prod_{i=1}^k \mu_i}\cdot P_0\tag{24}$$

Если подставить полученные соотношения в нормировочное условие (13), то получим:
$$P_0=\left(1+\sum_{k=1}^n\frac{\prod_{i=1}^k\lambda_i}{\prod_{i=1}^k \mu_i}\right)^{-1}\tag{25}$$

Полученные формулы для схемы рождения гибели очень полезны при решении простейших задач теории массового обслуживания. Однако, стохастические модели КИС, обычно имеют более сложную структуру. Для инженерных расчетов выходных параметров в марковских моделях более сложной структуры используются различные методы агрегирования, которые мы будем изучать в дальнейшем.